src/main/java/uk/gov/hmcts/reform/prl/services/cafcass/CaseDataService.java

```diff
@@ -136,10 +136,20 @@ public CafCassResponse getCaseData(String authorisation, String startDate, Strin
                     s2sToken,
                     cafCassSearchCaseTypeId
                 );
-                cafCassResponse = objectMapper.convertValue(
-                    searchResult,
-                    CafCassResponse.class
-                );
+                if (searchResult != null) {
+                    List<CafCassCaseDetail> cafCassCaseDetails = new ArrayList<>();
+                    searchResult.getCases().forEach(caseData -> {
+                        try {
+                            CafCassCaseDetail cafCassCaseDetail = objectMapper.convertValue(caseData, CafCassCaseDetail.class);
+                            cafCassCaseDetails.add(cafCassCaseDetail);
+                        } catch (IllegalArgumentException e) {
+                            log.info("Error while converting result case to Cafcass casedetails {}", caseData.getId());
+                        }
+                    });
+                    cafCassResponse.setCases(cafCassCaseDetails);
+                    cafCassResponse.setTotal(cafCassCaseDetails.size());
+                }
+
                 if (cafCassResponse.getCases() != null && !cafCassResponse.getCases().isEmpty()) {
                     log.info("CCD Search Result Size --> {}", cafCassResponse.getTotal());
```

Comment:
maybe either we should change this log to say Successful Cases result size or instead take the searchResult size total?